packages/block-library/src/embed/embed-placeholder.native.js

```diff
@@ -26,21 +37,37 @@ const EmbedPlaceholder = ( { icon, label, onFocus } ) => {
 	);
 
 	return (
-		<TouchableWithoutFeedback
-			accessibilityRole={ 'button' }
-			accessibilityHint={ __( 'Double tap to add a link.' ) }
-			onPress={ ( event ) => {
-				onFocus( event );
-			} }
-		>
-			<View style={ emptyStateContainerStyle }>
-				<View style={ styles.modalIcon }>{ icon }</View>
-				<Text style={ emptyStateTitleStyle }>{ label }</Text>
-				<Text style={ styles.emptyStateDescription }>
-					{ __( 'ADD LINK' ) }
-				</Text>
-			</View>
-		</TouchableWithoutFeedback>
+		<>
+			{ value ? (
+				<Text>{ value }</Text>
+			) : (
+				<TouchableWithoutFeedback
+					accessibilityRole={ 'button' }
+					accessibilityHint={ __( 'Double tap to add a link.' ) }
+					onPress={ ( event ) => {
+						onFocus( event );
+						setIsEmbedSheetVisible( true );
+					} }
+				>
+					<View style={ emptyStateContainerStyle }>
+						<View style={ styles.modalIcon }>{ icon }</View>
+						<Text style={ emptyStateTitleStyle }>{ label }</Text>
+						<Text style={ styles.emptyStateDescription }>
+							{ __( 'ADD LINK' ) }
+						</Text>
+					</View>
+				</TouchableWithoutFeedback>
+			) }
+			<EmbedBottomSheet
+				value={ value }
+				isVisible={ isEmbedSheetVisible }
+				onClose={ () => setIsEmbedSheetVisible( false ) }
+				onSetAttributes={ ( { url } ) => {
+					onChange( { target: { value: url } } );
```

Comment:
The object returned here has to match what is expected in the `edit` component:
https://github.com/WordPress/gutenberg/blob/8d3db371b033d4e331702a7a837978fac79e82ca/packages/block-library/src/embed/edit.js#L214

Reply:
Hmm, sounds like useful context to have as inline comment instead of just a PR comment, wdyt?

Reply:
Addressed with https://github.com/WordPress/gutenberg/pull/31343/commits/2a337a667a49b8919cf4f1b3b954daf37c2f9d7b.